INTMDB-587: New Feature: Backup Compliance Policy Support (#1127)
mongodb/terraform-provider-mongodbatlas: website/docs/r/backup_compliance_policy.html.markdown

}

resource "mongodbatlas_backup_compliance_policy" "backup_policy" {
  project_id                 = "<PROJECT-ID>"
  authorized_email           = "user@email.com"
  copy_protection_enabled    = false
  pit_enabled                = false
  encryption_at_rest_enabled = false

  restore_window_days = 7

  on_demand_policy_item {

    frequency_interval = 0
    frequency_type     = "ondemand"
    retention_unit     = "days"
    retention_value    = 3
  }
  
  scheduled_policy_items {
    
      frequency_interval = 6
      frequency_type     = "hourly"
      retention_unit     = "days"
      retention_value    = 7
    }

scheduled_policy_items {
    
      frequency_interval = 0
      frequency_type     = "daily"
      retention_unit     = "days"
      retention_value    = 7
    }

scheduled_policy_items {
    
      frequency_interval = 0
      frequency_type     = "weekly"
      retention_unit     = "weeks"
      retention_value    = 4
    }

scheduled_policy_items {
    
      frequency_interval = 0
      frequency_type     = "monthly"
      retention_unit     = "months"
      retention_value    = 12
    }

}
```

## Argument Reference

> * `project_id` - (Required) The unique identifier of the project for the Atlas cluster.

Reviewer: ```suggestion
* `project_id` - (Required) Unique 24-hexadecimal digit string that identifies your project
```

> retrieve.

Reviewer: ```suggestion

```

* `on_demand_policy_item.#.policy_item.#.frequency_interval` - The frequency interval for a set of snapshots.
* `on_demand_policy_item.#.policy_item.#.frequency_type` - A type of frequency (hourly, daily, weekly, monthly).
* `on_demand_policy_item.#.policy_item.#.retention_unit` - The unit of time in which snapshot retention is measured (days, weeks, months).
* `policies.#.policy_item.#.retention_value` - The number of days, weeks, or months the snapshot is retained.
* 
#### Scheduled Policy Items
* `scheduled_policy_items.#.policy_item` - A list of specifications for a policy.
* `scheduled_policy_items.#.policy_item.#.id` - Unique identifier for this policy item.
* `scheduled_policy_items.#.policy_item.#.frequency_interval` - The frequency interval for a set of snapshots.
* `scheduled_policy_items.#.policy_item.#.frequency_type` - A type of frequency (hourly, daily, weekly, monthly).
* `scheduled_policy_items.#.policy_item.#.retention_unit` - The unit of time in which snapshot retention is measured (days, weeks, months).
* `scheduled_policy_items.#.policy_item.#.retention_value` - The number of days, weeks, or months the snapshot is retained.

## Import

> Backup Compliance Policy entries can be imported using project project_id  in the format `PROJECTID`, e.g.

Reviewer: ```suggestion
Backup Compliance Policy entries can be imported using project project_id  in the format `project_id`, e.g.
```

```
$ terraform import mongodbatlas_backup_compliance_policy.backup_policy 5d0f1f73cf09a29120e173cf
```

> For more information see: [MongoDB Atlas API Reference.](hhttps://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Cloud-Backups/operation/updateDataProtectionSettings)

Reviewer: ```suggestion
For more information see: [MongoDB Atlas API Reference](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Cloud-Backups/operation/updateDataProtectionSettings) and [Backup Compliance Policy Prohibited Actions](https://www.mongodb.com/docs/atlas/backup/cloud-backup/backup-compliance-policy/#prohibited-actions)
```

---
layout: "mongodbatlas"
page_title: "MongoDB Atlas: backup_compliance_policy"
sidebar_current: "docs-mongodbatlas-resource-backup-compliance-policy"
description: |-
    Provides a Backup Compliance Policy resource.
---
# Resource: mongodbatlas_backup_compliance_policy

> `mongodbatlas_backup_compliance_policy` provides a resource that enables you to setup a Backup Compliance Policy resource. An Atlas Backup Compliance Policy configures the current protection policy settings for the project.

Reviewer: ```suggestion
`mongodbatlas_backup_compliance_policy` provides a resource that enables you to setup a Backup Compliance Policy resource. Prevent any user, regardless of role, from modifying or deleting specific cluster configurations and backups. When enabled, the Backup Compliance Policy will be applied as the minimum policy for all clusters and can only be disabled by contacting MongoDB support. Only supported for clusters M10 or higher.
```

> `mongodbatlas_backup_compliance_policy` provides a resource that enables you to setup a Backup Compliance Policy resource. Prevent any user, regardless of role, from modifying or deleting specific cluster configurations and backups. When enabled, the Backup Compliance Policy will be applied as the minimum policy for all clusters and can only be disabled by contacting MongoDB support. Only supported for clusters M10 or higher.

Reviewer: ```suggestion
`mongodbatlas_backup_compliance_policy` provides a resource that enables you to setup a Backup Compliance Policy resource. Prevent any user, regardless of role, from modifying or deleting specific cluster configurations and backups. When enabled, the Backup Compliance Policy will be applied as the minimum policy for all clusters and can only be disabled by contacting MongoDB support. Only supported for clusters M10 or higher.

When enabled, Backup Compliance Policy will be applied as the minimum backup policy to all clusters in your Project and will protect all existing snapshots. This will prevent any user, regardless of role, from modifying or deleting existing snapshots prior to expiration. Changes made to existing backup compliance policies will only apply to future snapshots.
```

-> **NOTE:** Groups and projects are synonymous terms. You may find `groupId` in the official documentation.

> -> **NOTE:** If you enable a Backup Compliance Policy, no user, regardless of role, can disable the Backup Compliance Policy without contacting MongoDB support, delete a backup snapshot, decrease the retention time for a snapshot after it's taken, disable Cloud Backup, modify the backup policy for an individual cluster below the minimum requirements set in the Backup Compliance Policy, or delete the Atlas project if any snapshots exist. For more details see (Backup Compliance Policy Prohibited Actions and Considerations)[https://www.mongodb.com/docs/atlas/backup/cloud-backup/backup-compliance-policy/#configure-a-backup-compliance-policy]. 

Reviewer: ```suggestion
-> **NOTE:** If you enable a Backup Compliance Policy, no user, regardless of role, can disable the Backup Compliance Policy without contacting MongoDB support, delete a backup snapshot, decrease the retention time for a snapshot after it's taken, disable Cloud Backup, modify the backup policy for an individual cluster below the minimum requirements set in the Backup Compliance Policy, or delete the Atlas project if any snapshots exist. For full list of impacts and more details see [Backup Compliance Policy Prohibited Actions and Considerations](https://www.mongodb.com/docs/atlas/backup/cloud-backup/backup-compliance-policy/#configure-a-backup-compliance-policy).
```

---
layout: "mongodbatlas"
page_title: "MongoDB Atlas: backup_compliance_policy"
sidebar_current: "docs-mongodbatlas-resource-backup-compliance-policy"
description: |-
    Provides a Backup Compliance Policy resource.
---
# Resource: mongodbatlas_backup_compliance_policy

> `mongodbatlas_backup_compliance_policy` provides a resource that enables you to setup a Backup Compliance Policy resource. [Backup Compliance Policy ](https://www.mongodb.com/docs/atlas/backup/cloud-backup/backup-compliance-policy) prevents any user, regardless of role, from modifying or deleting specific cluster settings, backups, and backup configurations. When enabled, the Backup Compliance Policy will be applied as the minimum policy for all clusters and backups in the project. It can only be disabled by contacting MongoDB support. This feature is only supported for cluster tiers M10+.

Reviewer 2: Minor fix
```suggestion
`mongodbatlas_backup_compliance_policy` provides a resource that enables you to set up a Backup Compliance Policy resource. [Backup Compliance Policy ](https://www.mongodb.com/docs/atlas/backup/cloud-backup/backup-compliance-policy) prevents any user, regardless of role, from modifying or deleting specific cluster settings, backups, and backup configurations. When enabled, the Backup Compliance Policy will be applied as the minimum policy for all clusters and backups in the project. It can only be disabled by contacting MongoDB support. This feature is only supported for cluster tiers M10+.
```

> When enabled, the Backup Compliance Policy will be applied as the minimum backup policy to all clusters in a Project and will protect all existing snapshots. This will prevent any user, regardless of role, from modifying or deleting existing snapshots prior to expiration. Changes made to existing backup policies will only apply to future snapshots.

Reviewer: ```suggestion

When enabled, the Backup Compliance Policy will be applied as the minimum backup policy to all clusters in a project and will protect all existing snapshots. This will prevent any user, regardless of role, from modifying or deleting existing snapshots prior to expiration. Changes made to existing backup policies will only apply to future snapshots.
```

Reviewer 2: Nit: lowercase "project"

> -> **NOTE:** Groups and projects are synonymous terms. You may find `groupId` in the official documentation.

Reviewer 2: ```suggestion
-> **NOTE:** Groups and projects are synonymous terms. You might find `groupId` in the official documentation.
```

-> **IMPORTANT NOTE:** Once enable a Backup Compliance Policy, no user, regardless of role, can disable the Backup Compliance Policy via Terraform, or any other method, without contacting MongoDB support.   This means that once enabled some resources defined in Terraform will no longer be modifiable.   See the full list of [Backup Compliance Policy Prohibited Actions and Considerations](https://www.mongodb.com/docs/atlas/backup/cloud-backup/backup-compliance-policy/#configure-a-backup-compliance-policy)

> 

Reviewer: ```suggestion

-> **NOTE:** With Backup Compliance Policy enabled, cluster backups are retained after a cluster is deleted and backups can be used normally until retention expiration. When the Backup Compliance Policy is not enabled, Atlas deletes the cluster's associated backup snapshots when a cluster is terminated. By default, a Backup Compliance Policy is not enabled. For more details see [Back Up, Restore, and Archive Data](https://www.mongodb.com/docs/atlas/backup-restore-cluster/). 

```

Reviewer 2: Suggest (on top of Zuhair's suggestion) removing "the default" and adding that as its own sentence:

When the Backup Compliance Policy is not enabled, Atlas deletes the cluster's associated backup snapshots when a cluster is terminated. By default, a Backup Compliance Policy is not enabled."

* `updated_user` - Email address that identifies the user who updated the Backup Compliance Policy settings. MongoDB Cloud ignores this email setting when you enable or update the Backup Compliance Policy settings.

### On Demand Policy Item
* `id` - Unique identifier of the backup policy item.
* `frequency_type` - Frequency associated with the backup policy item. For hourly policies, the frequency type is defined as `ondemand`. Note that this is a read-only value and not required in plan files - its value is implied from the policy resource type.
* `frequency_interval` - Desired frequency of the new backup policy item specified by `frequency_type` (hourly in this case). The supported values for hourly policies are `1`, `2`, `4`, `6`, `8` or `12` hours. Note that `12` hours is the only accepted value for NVMe clusters.
* `retention_unit` - Scope of the backup policy item: `days`, `weeks`, or `months`.
* `retention_value` - Value to associate with `retention_unit`.

> * 

Reviewer: ```suggestion

```

Reviewer: deleting extra bullet

* `retention_unit` - Scope of the backup policy item: `days`, `weeks`, or `months`.
* `retention_value` - Value to associate with `retention_unit`. Weekly policy must have retention of at least 7 days or 1 week. Note that for less frequent policy items, Atlas requires that you specify a retention period greater than or equal to the retention period specified for more frequent policy items. For example: If the daily policy item specifies a retention of two weeks, the weekly retention policy must specify two weeks or greater.

### Policy Item Monthly
* `id` - Unique identifier of the backup policy item.
* `frequency_type` - Frequency associated with the backup policy item. For monthly policies, the frequency type is defined as `monthly`. Note that this is a read-only value and not required in plan files - its value is implied from the policy resource type.
* `frequency_interval` - Desired frequency of the new backup policy item specified by `frequency_type` (monthly in this case). The supported values for weekly policies are 

>   * `1` through `28` where the number represents the day of the month i.e. `1` is the first of the month and `5` is the fifth day of the month.

Reviewer 2: Minor
```suggestion
  * `1` through `28` where the number represents the day of the month (i.e. `1` is the first of the month and `5` is the fifth day of the month).
```

  * `40` represents the last day of the month (depending on the month).
* `retention_unit` - Scope of the backup policy item: `days`, `weeks`, or `months`.
* `retention_value` - Value to associate with `retention_unit`. Monthly policy must have retention days of at least 31 days or 5 weeks or 1 month. Note that for less frequent policy items, Atlas requires that you specify a retention period greater than or equal to the retention period specified for more frequent policy items. For example: If the weekly policy item specifies a retention of two weeks, the montly retention policy must specify two weeks or greater.

## Import

Backup Compliance Policy entries can be imported using project project_id  in the format `project_id`, e.g.

```
$ terraform import mongodbatlas_backup_compliance_policy.backup_policy 5d0f1f73cf09a29120e173cf
```

> For more information see: [MongoDB Atlas API Reference](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Cloud-Backups/operation/updateDataProtectionSettings) and [Backup Compliance Policy Prohibited Actions](https://www.mongodb.com/docs/atlas/backup/cloud-backup/backup-compliance-policy/#prohibited-actions)

Reviewer 2: Minor
```suggestion
For more information see: [MongoDB Atlas API Reference](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Cloud-Backups/operation/updateDataProtectionSettings) and [Backup Compliance Policy Prohibited Actions](https://www.mongodb.com/docs/atlas/backup/cloud-backup/backup-compliance-policy/#prohibited-actions).
```

Author: Fixed

---
layout: "mongodbatlas"
page_title: "MongoDB Atlas: backup_compliance_policy"
sidebar_current: "docs-mongodbatlas-resource-backup-compliance-policy"
description: |-
    Provides a Backup Compliance Policy resource.
---
# Resource: mongodbatlas_backup_compliance_policy

`mongodbatlas_backup_compliance_policy` provides a resource that enables you to setup a Backup Compliance Policy resource. Prevent any user, regardless of role, from modifying or deleting specific cluster configurations and backups. When enabled, the Backup Compliance Policy will be applied as the minimum policy for all clusters and can only be disabled by contacting MongoDB support. Only supported for clusters M10 or higher.

> When enabled, Backup Compliance Policy will be applied as the minimum backup policy to all clusters in your Project and will protect all existing snapshots. This will prevent any user, regardless of role, from modifying or deleting existing snapshots prior to expiration. Changes made to existing backup compliance policies will only apply to future snapshots.

Reviewer 3: Needs a little editing - could probably be paired down but we can work with docs on a refresh going forward.
```suggestion
`mongodbatlas_backup_compliance_policy` provides a resource that enables you to setup a Backup Compliance Policy resource. [Backup Compliance Policy ](https://www.mongodb.com/docs/atlas/backup/cloud-backup/backup-compliance-policy) prevents any user, regardless of role, from modifying or deleting specific cluster settings, backups, and backup configurations. When enabled, the Backup Compliance Policy will be applied as the minimum policy for all clusters and backups in the project. It can only be disabled by contacting MongoDB support. This feature is only supported for cluster tiers M10+.

When enabled, the Backup Compliance Policy will be applied as the minimum backup policy to all clusters in a Project and will protect all existing snapshots. This will prevent any user, regardless of role, from modifying or deleting existing snapshots prior to expiration. Changes made to existing backup policies will only apply to future snapshots.  
```

Author: Edited

-> **NOTE:** Groups and projects are synonymous terms. You may find `groupId` in the official documentation.

> -> **NOTE:** If you enable a Backup Compliance Policy, no user, regardless of role, can disable the Backup Compliance Policy without contacting MongoDB support, delete a backup snapshot, decrease the retention time for a snapshot after it's taken, disable Cloud Backup, modify the backup policy for an individual cluster below the minimum requirements set in the Backup Compliance Policy, or delete the Atlas project if any snapshots exist. For full list of impacts and more details see [Backup Compliance Policy Prohibited Actions and Considerations](https://www.mongodb.com/docs/atlas/backup/cloud-backup/backup-compliance-policy/#configure-a-backup-compliance-policy).

Reviewer 3: This is important so I'd keep it short - basically once enabled you can't undo it with Terraform.  Something like: 
```suggestion
-> **IMPORTANT NOTE:** Once enable a Backup Compliance Policy, no user, regardless of role, can disable the Backup Compliance Policy via Terraform, or any other method, without contacting MongoDB support.   This means that once enabled some resources defined in Terraform will no longer be modifiable.   See the full list of [Backup Compliance Policy Prohibited Actions and Considerations](https://www.mongodb.com/docs/atlas/backup/cloud-backup/backup-compliance-policy/#configure-a-backup-compliance-policy).  
```

## Example Usage

```terraform
resource "mongodbatlas_cluster" "my_cluster" {
  project_id   = "<PROJECT-ID>"
  name         = "clusterTest"

>   disk_size_gb = 5

Reviewer 3: We quit including this optional parameter in examples because when the default disk sizes changes it breaks all examples.  
```suggestion
```

Reviewer: makes sense, in the case going to create a separate PR removing this parameter from all our examples

  restore_window_days      = 4

  policies {
    id = mongodbatlas_cluster.my_cluster.snapshot_backup_policy.0.policies.0.id

    policy_item {
      id                 = mongodbatlas_cluster.my_cluster.snapshot_backup_policy.0.policies.0.policy_item.0.id
      frequency_interval = 1
      frequency_type     = "hourly"
      retention_unit     = "days"
      retention_value    = 1
    }
    policy_item {
      id                 = mongodbatlas_cluster.my_cluster.snapshot_backup_policy.0.policies.0.policy_item.1.id
      frequency_interval = 1
      frequency_type     = "daily"
      retention_unit     = "days"
      retention_value    = 2
    }
    policy_item {
      id                 = mongodbatlas_cluster.my_cluster.snapshot_backup_policy.0.policies.0.policy_item.2.id
      frequency_interval = 4
      frequency_type     = "weekly"
      retention_unit     = "weeks"
      retention_value    = 3
    }
    policy_item {
      id                 = mongodbatlas_cluster.my_cluster.snapshot_backup_policy.0.policies.0.policy_item.3.id
      frequency_interval = 5
      frequency_type     = "monthly"
      retention_unit     = "months"
      retention_value    = 4
    }
  }

> }

Reviewer 3: This is a deprecated resource, we should only use current resources - in this case `mongodbatlas_cloud_backup_schedule`

data "mongodbatlas_cloud_provider_snapshot_backup_policy" "test" {
  project_id   = mongodbatlas_cloud_provider_snapshot_backup_policy.test.project_id
  cluster_name = mongodbatlas_cloud_provider_snapshot_backup_policy.test.cluster_name

> }

Reviewer 3: Also deprecated, need to replace with `mongodbatlas_cloud_backup_schedule`

data "mongodbatlas_backup_compliance_policy" "backup_policy" {
  project_id = mongodbatlas_backup_compliance_policy.backup_policy.id
}

resource "mongodbatlas_backup_compliance_policy" "backup_policy" {
  project_id                 = "<PROJECT-ID>"
  authorized_email           = "user@email.com"
  copy_protection_enabled    = false
  pit_enabled                = false
  encryption_at_rest_enabled = false

  restore_window_days = 7

  on_demand_policy_item {

    frequency_interval = 0
    frequency_type     = "ondemand"
    retention_unit     = "days"
    retention_value    = 3
  }
  
  scheduled_policy_items {
    
      frequency_interval = 6
      frequency_type     = "hourly"
      retention_unit     = "days"
      retention_value    = 7

>     }

Reviewer 3: So this is a pretty big change from how easy we made `mongodbatlas_cloud_backup_schedule`.  In that we crated frequency type blocks since we knew there would really only be one policy - which this makes easy as well.  However it's now two different ways to do the same thing I think and will likely prove confusing for users.  Is there any reason we went down a different path here I'm not seeing?  (haven't looked at code yet but just from the docs this is pretty different).

scheduled_policy_items {
    
      frequency_interval = 0
      frequency_type     = "daily"
      retention_unit     = "days"
      retention_value    = 7
    }

scheduled_policy_items {
    
      frequency_interval = 0
      frequency_type     = "weekly"
      retention_unit     = "weeks"
      retention_value    = 4
    }

scheduled_policy_items {
    
      frequency_interval = 0
      frequency_type     = "monthly"
      retention_unit     = "months"
      retention_value    = 12
    }

}
```

## Argument Reference

* `project_id` - (Required) Unique 24-hexadecimal digit string that identifies your project

## Attributes Reference

> In addition to all arguments above, the following attributes are exported:

Reviewer 3: This contains both the argument reference and attributes reference, i.e. exported items.  We need to separate these out.

Reviewer: updated

> * `authorized_email` - (Email address of the user who authorized to updated the Backup Compliance Policy settings.

Reviewer 3: So the source docs are same - so looks like a docs bug?  This doesn't quite make sense.  Assuming this is the email address of the authorized party but we should check, i.e.:
```suggestion
* `authorized_email` - Email address of a security or legal representative for the Backup Compliance Policy who is authorized to update the Backup Compliance Policy settings.   
```

Reviewer: Yes the Atlas UI also prompts users for an authorized email 

<img width="974" alt="image" src="https://user-images.githubusercontent.com/47370943/233239396-09a1e2ed-be80-4790-bf73-70972438473b.png">

Reviewer: team confirmed this is doc bug, fix is coming as part of https://jira.mongodb.org/browse/HELP-44876

* `copy_protection_enabled` - Flag that indicates whether to enable additional backup copies for the cluster. If unspecified, this value defaults to false.
* `pit_enabled` - Flag that indicates whether the cluster uses Continuous Cloud Backups with a Backup Compliance Policy. If unspecified, this value defaults to false.
* `encryption_at_rest_enabled` - Flag that indicates whether Encryption at Rest using Customer Key Management is required for all clusters with a Backup Compliance Policy. If unspecified, this value defaults to false.

> * `reference_minute_of_hour` - UTC Minute of day between 0 and 59 representing which minute of the referenceHourOfDay that Atlas takes the snapshot.

Reviewer 3: This is a parameter of `mongodbatlas_cloud_backup_schedule`, as is reference_hour_of_day (not here).  But these are not in the backup compliancy policy docs page (https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Cloud-Backups/operation/updateDataProtectionSettings).  So this is either a docs bug in the core docs or a docs bug here.   I would expect both of these to be in compliance policy though

Author: Removed reference_minute_of_hour as this is not in backup compliance policy structure at the moment

* `updated_date` - ISO 8601 timestamp format in UTC that indicates when the user updated the Data Protection Policy settings. MongoDB Cloud ignores this setting when you enable or update the Backup Compliance Policy settings.
* `updated_user` - Email address that identifies the user who updated the Backup Compliance Policy settings. MongoDB Cloud ignores this email setting when you enable or update the Backup Compliance Policy settings.

### On Demand Policy Item
* `on_demand_policy_item.0.policy_item` - A list of specifications for a policy.
* `on_demand_policy_item.#.policy_item.#.id` - Unique identifier for this policy item.
* `on_demand_policy_item.#.policy_item.#.frequency_interval` - The frequency interval for a set of snapshots.
* `on_demand_policy_item.#.policy_item.#.frequency_type` - A type of frequency (hourly, daily, weekly, monthly).
* `on_demand_policy_item.#.policy_item.#.retention_unit` - The unit of time in which snapshot retention is measured (days, weeks, months).
* `policies.#.policy_item.#.retention_value` - The number of days, weeks, or months the snapshot is retained.
* 
#### Scheduled Policy Items

> * `scheduled_policy_items.#.policy_item` - A list of specifications for a policy.

Reviewer 3: That # is why we did the much more clean policy_item_[frequencyType] {} blocks.   I don't think this will work for users.

  project_id   = mongodbatlas_cloud_backup_schedule.test.project_id
  cluster_name = mongodbatlas_cloud_backup_schedule.test.cluster_name
}

data "mongodbatlas_backup_compliance_policy" "backup_policy" {
  project_id = mongodbatlas_cloud_backup_schedule.test.id
}

resource "mongodbatlas_backup_compliance_policy" "backup_policy" {
  project_id                 = "<PROJECT-ID>"
  authorized_email           = "user@email.com"
  copy_protection_enabled    = false
  pit_enabled                = false
  encryption_at_rest_enabled = false

  restore_window_days = 7

  on_demand_policy_item {
		  frequency_interval = 0
		  retention_unit     = "days"
		  retention_value    = 3
		}
		
		policy_item_hourly {
			frequency_interval = 6
			retention_unit     = "days"
			retention_value    = 7
		  }
	  
		policy_item_daily {
			frequency_interval = 0
			retention_unit     = "days"
			retention_value    = 7
		  }
	  
		  policy_item_weekly {
			frequency_interval = 0
			retention_unit     = "weeks"
			retention_value    = 4
		  }
	  
		  policy_item_monthly {
			frequency_interval = 0
			retention_unit     = "months"
			retention_value    = 12
		  }

}
```

## Argument Reference

> * `project_id` - (Required) Unique 24-hexadecimal digit string that identifies your project

Reviewer: ```suggestion
* `project_id` - (Required) Unique 24-hexadecimal digit string that identifies your project.
```

  * `40` represents the last day of the month (depending on the month).
* `retention_unit` - Scope of the backup policy item: `days`, `weeks`, or `months`.
* `retention_value` - Value to associate with `retention_unit`. Monthly policy must have retention days of at least 31 days or 5 weeks or 1 month. Note that for less frequent policy items, Atlas requires that you specify a retention period greater than or equal to the retention period specified for more frequent policy items. For example: If the weekly policy item specifies a retention of two weeks, the montly retention policy must specify two weeks or greater.

## Import

Backup Compliance Policy entries can be imported using project project_id  in the format `project_id`, e.g.

```
$ terraform import mongodbatlas_backup_compliance_policy.backup_policy 5d0f1f73cf09a29120e173cf
```

For more information see: [MongoDB Atlas API Reference](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Cloud-Backups/operation/updateDataProtectionSettings) and [Backup Compliance Policy Prohibited Actions](https://www.mongodb.com/docs/atlas/backup/cloud-backup/backup-compliance-policy/#prohibited-actions).

> 

Reviewer: ```suggestion
```
